cryptography/kzg_single_open/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use bls12_381::{multi_pairings, G1Point, G2Point, G2Prepared, Scalar};
+use kzg_multi_open::{commit_key::CommitKey, opening_key::OpeningKey};
+use polynomial::{domain::Domain, monomial::PolyCoeff};
+
+pub struct Proof {
+    pub quotient_commitment: bls12_381::G1Point,
+    pub claimed_evaluation: Scalar,
+}
+
+/// Takes a polynomial in lagrange form in reverse bit order and an input point
+/// and computes a proof that the polynomial is correctly evaluated at the input point.
+///
+// Note: The total time taken is around 40-50ms on a single thread. 2-3ms of this is
+// division, fft and poly_eval. The rest of it is committing.
+pub fn compute_proof(
+    commit_key: &CommitKey,
+    domain: &Domain,
+    polynomial_lagrange: &[Scalar],
+    input_point: Scalar,
+) -> Proof {
+    // Bit reverse the polynomial and interpolate it.
+    //
+    // The bit-reversal is an artifact of a feature we want to maintain
+    // when we use FK20.
+    let mut poly_lagrange = polynomial_lagrange.to_vec();
+    reverse_bit_order(&mut poly_lagrange);
+    let polynomial_coeff = domain.ifft_scalars(poly_lagrange);
+
+    let quotient_poly = divide_by_linear(&polynomial_coeff, input_point);
+    let quotient_commitment = commit_key.commit_g1(quotient_poly.as_slice());
+    let claimed_evaluation = poly_eval(&polynomial_coeff, &input_point);
```

Comment:
None of these methods are actually new -- they are all a part of eip7594 specs, so in a way this does not add any new complexity

Reply:
- divide_by_linear is a simpler version of `divide_polynomialcoeff` where b is a linear polynomial
- poly_eval is `evaluate_polynomial_coeff`
- commit_g1 is g1_lincomb